subsys/bluetooth/host/mesh/Kconfig

```diff
@@ -11,6 +11,7 @@ menuconfig BT_MESH
 	select TINYCRYPT
 	select TINYCRYPT_AES
 	select TINYCRYPT_AES_CMAC
+	select BT_OBSERVER
```

Comment:
Most mesh app currently have this in their `prj.conf`:

```
CONFIG_BT_OBSERVER=y
CONFIG_BT_BROADCASTER=y
```

If you're going to add this `select` statement, shouldn't you also add `BROADCASTER`? And if you add these, shouldn't you remove them from the respective `prj.conf` files?

Reply:
OK, I'll add another patch for this

Reply:
That said, is it the right thing to add these? There's a general aversion to using select, and it should be limited to very specific cases. @carlescufi do you think this qualifies?

Reply:
@MariuszSkamra what exactly was it that prompted you to add the select statement? It seems there's a bug in the Kconfig either way: if we don't have select there it's then at least missing a depends on for both observer and broadcaster.

Reply:
@jhedberg I believe, one of the samples failed when I ran sanitycheck.

Reply:
I'm running to see which one.
Nevertheless, I think that having those selected in Kconfig is logical. Mesh device is broadcaster and observer imo. Having GATT bearer enforces peripheral role, provisioner over GATT bearer enforces central role etc.

Reply:
@jhedberg 
`
CONFIG_BT_OBSERVER=y
CONFIG_BT_BROADCASTER=y
`
is missing in samples/bluetooth/mesh_demo/prj.conf

Reply:
@MariuszSkamra for the record, the rationale for preferring `depends on` and avoiding `select` (and when exceptions can be made) is outlined here:

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/kbuild/kconfig-language.txt#n128

Reply:
Yep, I completely agree with @jhedberg here. We should add a `depends on` for both instead of the current `select` statements. I see no justification in this case to use `select` other than convenience for the user, but this might backfire later as `select` is dangerous long-term.

Reply:
@jhedberg @carlescufi Thanks, I see now. You have right, the dependency is the right way to do it. I was not aware that 'select' don't resolve the dependencies.